js/tidex2.js

```diff
@@ -0,0 +1,960 @@
+        const side = this.safeString (trade, 'type');
+        const price = this.safeString (trade, 'price');
+        const amount = this.safeString (trade, 'amount');
+        const id = this.safeString (trade, 'tid');
+        const symbol = this.safeSymbol (undefined, market);
+        return this.safeTrade ({
+            'id': id,
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'order': undefined,
+            'type': undefined,
+            'takerOrMaker': undefined,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'fee': undefined,
+        }, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (since === undefined) {
+            since = 1;
+        }
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+            'since': since, // Min 1; Market History Since Selected tid
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // Default 50; min 1; max 1000
+        }
+        const response = await this.publicGetHistoryResult (this.extend (request, params));
+        //
+        //     [
+        //         {
+        //             "tid":135762344,
+        //             "date":1646294384,
+        //             "price":"21991.91",
+        //             "type":"buy",
+        //             "amount":"0.0024",
+        //             "total":"52.780584"
+        //         }
+        //     ]
+        //
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        //     {
+        //         "time":1646205840,
+        //         "open":"21290.48",
+        //         "close":"21290.48",
+        //         "highest":"21290.48",
+        //         "lowest":"21290.48",
+        //         "volume":"0",
+        //         "amount":"0",
+        //         "market":"YFI_USDT"
+        //     }
+        //
+        return [
+            this.safeTimestamp (ohlcv, 'time'),
+            this.safeNumber (ohlcv, 'open'),
+            this.safeNumber (ohlcv, 'highest'),
+            this.safeNumber (ohlcv, 'lowest'),
+            this.safeNumber (ohlcv, 'close'),
+            this.safeNumber (ohlcv, 'volume'),
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'interval': this.timeframes[timeframe],
+            'market': market['id'],
+        };
+        limit = (limit === undefined) ? 1501 : limit;
+        if (since === undefined) {
```

Comment:
because `since` is a required parameter on the exchange side for an OHLCV request, we need to throw an error if it is not provided see here https://github.com/ccxt/ccxt/blob/master/js/deribit.js#L915